ecommerce/core/templatetags/core_extras.py

```diff
@@ -31,6 +36,90 @@ def settings_value(name):
     return getattr(settings, name)
 
 
+@register.simple_tag(name='is_special_coupon')
+def is_special_coupon(object_passed):
+    """
+    Checks if the coupon name meets the format criteria of a special coupon. The coupon name is
+    extracted from the 'object_passed' argument which could be an Order, Basket or OrderDiscount instance.
+
+    Arguments:
+        object_passed (Order or SimpleLazyObject or OrderDiscount): Order of the purchase, its basket or its OrderDiscount.
+
+    Returns:
+        bool: True if the coupon name is a valid special coupon, False otherwise.
+    """
+    is_order = isinstance(object_passed, Order)
+    is_basket = isinstance(object_passed, SimpleLazyObject)
```

Comment:
Why is 'SimpleLazyObject' an instance of a Basket? @Jacatove

Reply:
I got his while debugging:
![image](https://user-images.githubusercontent.com/40271196/140320842-b86dc0cb-7375-4653-9d6d-9d770354b960.png)
Should I change the docstring or variable name to SimpleLazyObject? @Squirrel18